### PR TITLE
leo132 - add R kernel

### PR DESCRIPTION
**Pull request by ahaessly**
Add R kernel

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: jupyter-docker/Dockerfile [hunk 1]
```diff
@@ -105,12 +105,10 @@ RUN apt-get install -yq --no-install-recommends \
     python-tk \
     python-numpy \
     libz-dev \
- && apt-get clean \
- && rm -rf /var/lib/apt/lists/* \
 
  # NOTE! not sure why, but this must run before pip installation
  && useradd -m -s /bin/bash -N -u $UID $USER \
- # jessie's default pip doesn't work well with jupyter
+ # jessie's default pip doesn't work well with jupyter # what about stretch?
```
Comment (helgridly) [inline jupyter-docker/Dockerfile]: yeah maybe this comment is superfluous

Comment (helgridly) [reply jupyter-docker/Dockerfile]: in any case, it' confusing for the comment to refer to jessie when we're not on jessie

**Comment by ahaessly** [comment]
reverted the debian upgrade back to jessie
installing R version 3.3.1 from jessie-backport

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/NotebookInteractionSpec.scala [hunk 2]
```diff
@@ -175,5 +175,15 @@ class NotebookInteractionSpec extends FreeSpec with LeonardoTestUtils with Befor
       }
     }
 
+    "should create a notebook with a working R kernel and import installed packages" in withWebDriver { implicit driver =>
+      Orchestration.billing.addUserToBillingProject(billingProject.value, ronEmail, Orchestration.billing.BillingProjectRole.User)(hermioneAuthToken)
+
+      withNewNotebook(ronCluster, RKernel) { notebookPage =>
+        notebookPage.executeCell("library(SparkR)").get should include("SparkR")
+        notebookPage.executeCell("sparkR.session()")
```
Comment (ansingh7115) [inline automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/NotebookInteractionSpec.scala]: Maybe also run a small spark job?

Comment (rtitle) [reply automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/NotebookInteractionSpec.scala]: +1

File: jupyter-docker/Dockerfile [hunk 3]
```diff
@@ -146,4 +165,4 @@ USER $USER
 WORKDIR $HOME
 
 EXPOSE $JUPYTER_PORT
-ENTRYPOINT ["pyspark"]
+ENTRYPOINT ["jupyter notebook"]
```
Comment (rtitle) [inline jupyter-docker/Dockerfile]: This PR probably needs a rebase; this change is in dev now.

File: jupyter-docker/Dockerfile [hunk 4]
```diff
@@ -129,6 +127,27 @@ RUN apt-get install -yq --no-install-recommends \
  && pip install matplotlib-venn \
  && pip install "https://github.com/all-of-us/workbench/archive/pyclient-v0-1-rc3.zip#egg=aou_workbench_client&subdirectory=client/py"
 
+#######################
+# R Kernel
+#######################
+
+RUN apt-get -t jessie-backports install -y --no-install-recommends \
```
Comment (rtitle) [inline jupyter-docker/Dockerfile]: It'd be good to do all these in the same layer (`RUN` command)`. E.g. similar to how python does the apt-gets and pips separated by &&.

**Comment by ahaessly** [comment]
jenkins retest

**Comment by ahaessly** [comment]
jenkins retest

**Pull request merged**